x-pack/test/functional/apps/upgrade_assistant/overview_page.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+export default function upgradeAssistantOverviewPageFunctionalTests({
+  getService,
+  getPageObjects,
+}: FtrProviderContext) {
+  const PageObjects = getPageObjects(['upgradeAssistant', 'common']);
+  const retry = getService('retry');
+  const security = getService('security');
+  const testSubjects = getService('testSubjects');
+  const es = getService('es');
+
+  describe('Overview Page', function () {
+    this.tags('skipFirefox');
+
+    before(async () => {
+      await security.testUser.setRoles(['superuser', 'global_upgrade_assistant_role']);
+    });
+
+    after(async () => {
+      await security.testUser.restoreDefaults();
+    });
+
+    beforeEach(async () => {
+      await PageObjects.upgradeAssistant.navigateToPage();
+
+      await retry.waitFor('Upgrade Assistant overview page to be visible', async () => {
+        return testSubjects.exists('overview');
+      });
+    });
+
+    it('Should render all steps', async () => {
+      testSubjects.exists('backupStep-incomplete');
+      testSubjects.exists('fixIssuesStep-incomplete');
+      testSubjects.exists('fixLogsStep-incomplete');
+      testSubjects.exists('upgradeStep');
+    });
+
+    describe('fixLogsStep', () => {
+      beforeEach(async () => {
+        // Access to system indices will be deprecated and should generate a deprecation log
+        await es.transport.request({
+          method: 'GET',
+          path: '/.apm-agent-configuration',
+        });
+
+        await PageObjects.upgradeAssistant.navigateToPage();
+        await PageObjects.upgradeAssistant.clickDeprecationLoggingToggle();
+
+        await retry.waitFor('UA external links title to be present', async () => {
+          return testSubjects.isDisplayed('externalLinksTitle');
+        });
+      });
+
+      it('Shows warnings callout if there are deprecations', async () => {
+        testSubjects.exists('hasWarningsCallout');
+      });
+
+      it('Shows shows no warnings callout if there are no deprecations', async () => {
+        await PageObjects.upgradeAssistant.clickResetLastCheckpointButton();
```

Comment:
Looks like this is what's failing on CI. Is this working locally for you?

Reply:
Yeah they all pass locally, running them with `node scripts/functional_test_runner.js --config test/functional/config.js --grep="Upgrade Assistant"` will try running them all instead, see if that makes it fail. Otherwise I'll rethink the test a bit

Reply:
Turns out I had a beforeEach that was clicking the deprecation logs toggle, for the first test it was enabled but when reaching the second it would disable it and the test was waiting for a UI state that was never going to be shown. It's very strange that the tests always pass locally 🤷🏼. Fixed with: https://github.com/elastic/kibana/pull/111370/commits/f0a0d1c06dcb6e7f04c51859944bd8611454a6d0